internal/transport/http_util.go

```diff
@@ -120,8 +118,32 @@ type decodeState struct {
 	statsTags      []byte
 	statsTrace     []byte
 	contentSubtype string
+}
+
+func newDecodeState(serverSide bool, ignoreContentType bool) *decodeState {
+	return &decodeState{
+		serverSide:        serverSide,
+		ignoreContentType: ignoreContentType,
+	}
+}
+
+// Records the states during HPACK decoding. Must be reset once the
+// decoding of the entire headers are finished.
```

Comment:
Is this "reset" comment correct?  Do we ever reset it? I thought we just make new ones every time.

Reply:
Deleted the reset. I don't think reset is done, we just create a new struct every time. This is old comments from before.